tests/cstor/pool/negative/provision_without_pool.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package negative
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/openebs/maya/cmd/cstor-pool-mgmt/pool"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	"github.com/openebs/maya/tests"
+	"github.com/openebs/maya/tests/cstor"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("[cstor] [-ve] TEST PROVISION WITHOUT POOL", func() {
+	var (
+		err error
+	)
+
+	BeforeEach(func() {
+		By("creating storagepoolclaim")
+		spcConfig := &tests.SPCConfig{
+			Name:               spcName,
+			DiskType:           "sparse",
+			PoolCount:          cstor.PoolCount,
+			IsOverProvisioning: false,
+			PoolType:           "striped",
+		}
+		ops.Config = spcConfig
+		spcObj = ops.BuildAndCreateSPC()
+		By("Creating SPC, Desired Number of CSP Should Be Created", func() {
+			ops.VerifyDesiredCSPCount(spcObj, cstor.PoolCount)
+		})
+	})
+
+	AfterEach(func() {
+		By("deleting storagepoolclaim")
+
+		err = ops.SPCClient.Delete(spcObj.Name, &metav1.DeleteOptions{})
+		Expect(err).To(BeNil(), "while deleting the storagepoolclaim {%s}", spcObj.Name)
+
+		cspCount := ops.GetCSPCount(getLabelSelector(spcObj))
+		Expect(cspCount).To(Equal(0), "stale CSP")
+	})
+
+	When("creating storagepoolclaim without", func() {
+		It("Testing SPC deletion if pool does not exist", func() {
+			spcLabel := getLabelSelector(spcObj)
+			poolPodList, err := ops.PodClient.
+				WithNamespace(openebsNamespace).
+				List(metav1.ListOptions{LabelSelector: spcLabel})
+			Expect(err).To(BeNil())
+			Expect(len(poolPodList.Items)).Should(BeNumerically(">=", 1))
+
+			tPod := poolPodList.Items[0]
+			var puid string
+			for _, e := range tPod.Spec.Containers[0].Env {
+				if e.Name == "OPENEBS_IO_CSTOR_ID" {
+					puid = e.Value
+				}
+			}
+			Expect(len(puid)).Should(BeNumerically(">=", 1))
+
+			cspList, err := ops.CSPClient.List(metav1.ListOptions{LabelSelector: string(apis.StoragePoolClaimCPK) + "=" + spcObj.Name})
+			Expect(err).To(BeNil())
+			Expect(len(cspList.Items)).Should(BeNumerically(">=", 1))
+			var cspObj apis.CStorPool
+			for _, l := range cspList.Items {
+				if l.GetUID() == types.UID(puid) {
+					cspObj = l
+					break
+				}
+			}
+
```

Comment:
If I got you right, the above code is trying to find a matching CSP of pool pod.
Why can't we get one CSP and figure out the corresponding pod via the label.

Reply:
yes. We can do that also.